### PR TITLE
Extend validate to check for valid members

### PR DESCRIPTION
**Pull request by chris0x44**
Extend validation to check that members are not just removed and potentially detect errors due to type changes.

### CODE REVIEW

File: labs/memory_bound/data_packing/solution.h [hunk 1]
```diff
@@ -15,7 +15,11 @@ struct S {
   bool b;
 
   bool operator<(const S &s) const { return this->i < s.i; }
+  long long get_l() const { return l; }
```
Comment (dendibakh) [inline labs/memory_bound/data_packing/solution.h:18]: I think we can get away without accessor methods.

File: labs/memory_bound/data_packing/validate.cpp [hunk 2]
```diff
@@ -3,6 +3,49 @@
...
+  std::cerr << "Validation Failed. Value " << var_name << " is " << received
+            << ". Expected is " << expected << " for intialization values "
+            << first_value << " and " << second_value << std::endl;
+}
+
+bool check_entry(int first_value, int second_value) {
+  S entry = create_entry(first_value, second_value);
+
+  if (entry.i != first_value) {
+    report_failure("i", entry.i, first_value, first_value, second_value);
```
Comment (dendibakh) [inline labs/memory_bound/data_packing/validate.cpp:19]: minor: maybe we can avoid code duplication...

**Review by dendibakh — COMMENTED** [review]
In general LGTM, but please address a few minor issues.

File: labs/memory_bound/data_packing/solution.h (hunk 1 above)
Comment (chris0x44) [reply labs/memory_bound/data_packing/solution.h:18]: The idea of placing accessors in the struct was to isolate `validate.cpp` from changes in order to solve this challenge. This way, the only place the user needs to adapt code is in the `init.cpp` and `solution.*` files. Alternative approach could be to provide free functions as accessors but that wouldn't really make a big difference. What do you think?

File: labs/memory_bound/data_packing/validate.cpp (hunk 2 above)
Comment (chris0x44) [reply labs/memory_bound/data_packing/validate.cpp:19]: Agreed. Will change.

**Review by dendibakh — APPROVED** [review]

**Pull request merged**